test/buyer.test.js

```diff
@@ -97,4 +96,43 @@ describe("testing buyer signup",() =>{
       expect(response.statusCode).toBe(404);
   });
 
-}); 
+  it('should log out user and clear token cookie', async () => {
+    const response = await request
+      .get('/api/user/logout')
+      .set('Cookie', cookie);
+    expect(response.status).toBe(200);
+    expect(response.body.status).toBe('success');
+    expect(response.body.message).toBe('User logged out successfully');
+    expect(response.header['set-cookie']).toBeDefined();
+    expect(response.header['set-cookie'][0]).toContain('token=; Path=/; Expires=');
+  });
+
+  });
+
+  describe('/cart/add/:productId endpoint', () => {
+    test('should return 404 if product is not found', async () => {
+      const res = await request.post('/api/cart/add/400').set('token', `Bearer ${token}`);
+      expect(res.statusCode).toBe(404);
+    });
+  
+    test('should add a product to an existing cart', async () => {
+       const productId = 1;
+      const res = await request.post(`/api/cart/add/${productId}`).set('token',`Bearer ${token}`);
+      expect(res.statusCode).toBe(201);
+    });
+
+    test('should return 400 if product is already in cart', async () => {
+      const productId = 1;
+      const res = await request.post(`/api/cart/add/${productId}`)
+        .set('token', `Bearer ${token}`)
+      expect(res.statusCode).toBe(400);
+  })
+})
+
+  describe('/getCartItems endpoint', () => {
+    test('should return array of all items', async () => {
+        const res = await request.get('/api/cart/getAll').set('token',  `Bearer ${token}`);
+        expect(res.statusCode).toBe(304);
+       
+    });
+  })
```

Comment:
- write more tests for different scenarios that may happen when performing this action.